File: frontend-pedidos360/src/app/auth/roles.ts
// App Roles definidos en Microsoft Entra ID (claim "roles" del access token)
export const Roles = {
  ADMIN: 'ROLE_ADMINISTRADOR',
  OPERADOR: 'ROLE_OPERADOR',
  CLIENTE: 'ROLE_CLIENTE',
} as const;

export type Role = (typeof Roles)[keyof typeof Roles];

export const ROLE_LABELS: Record<Role, string> = {
  [Roles.ADMIN]: 'Administrador',
  [Roles.OPERADOR]: 'Operador',
  [Roles.CLIENTE]: 'Cliente',
};

File: ms-pedidos360-bff/src/main/java/com/pedidos360/bff/security/Roles.java
package com.pedidos360.bff.security;

// App Roles definidos en Microsoft Entra ID (llegan en el claim "roles" del access token)
public final class Roles {

    public static final String ADMIN = "ROLE_ADMINISTRADOR";
    public static final String OPERADOR = "ROLE_OPERADOR";
    public static final String CLIENTE = "ROLE_CLIENTE";

    // Expresiones para @PreAuthorize
    public static final String IS_ADMIN = "hasAuthority('" + ADMIN + "')";
    public static final String IS_STAFF = "hasAnyAuthority('" + ADMIN + "','" + OPERADOR + "')";
    public static final String IS_ANY = "hasAnyAuthority('" + ADMIN + "','" + OPERADOR + "','" + CLIENTE + "')";

    private Roles() {
    }
}
